api/models.py

```diff
@@ -338,8 +339,9 @@ def validate_params(cls, params: dict):
             return False, f"Invalid state value '{state}'"
 
         result = params.get('regression_data.result')
-        if result and result not in [result.value for result in ResultValues]:
-            return False, f"Invalid result value '{result}'"
+        if result and result != 'null':
```

Comment:
This seems way too intrusive, if we want to make `null` in the query become `None` then I believe we should be able to do this in a single place as an extension to what FastAPI is already doing when parsing the query parameters.